big-red-shuttle/EmergencyViewController.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  EmergencyViewController.swift
+//  big-red-shuttle
+//
+//  Created by Austin Astorga on 10/26/16.
+//  Copyright © 2016 cuappdev. All rights reserved.
+//
+
+import UIKit
+
+class EmergencyViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    
+    var tableView: UITableView!
+    let emergencyArray = [("Call Cornell Univeristy Police", 6072551111), ("Call Emergency Services", 911), ("Call Blue Light Escorts", 6072557373), ("Call Gannett Health Services", 6072555155)] //add more later
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Do any additional setup after loading the view.
+        self.title = "Emergency"
+        let tableViewFrame = CGRect(x: 0, y: 0, width: view.bounds.width, height: view.bounds.height - (tabBarController?.tabBar.frame.height)!)
+        
+        self.tableView = UITableView(frame: tableViewFrame, style: .grouped)
+        self.tableView.delegate = self
+        self.tableView.dataSource = self
+        tableView.tableFooterView = UIView()
+        tableView.backgroundColor = UIColor(red:0.96, green:0.96, blue:0.96, alpha:1.0)
+        tableView.contentInset = UIEdgeInsetsMake(-1.0, 0.0, 0.0, 0.0)
+        
+        self.view.addSubview(self.tableView)
+    }
+
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    /*phoneNumberToFormattedString takes in an integer and converts it to a (XXX) YYY-YYYY string */
+    func phoneNumberToFormattedString(phoneNumber: Int) -> String {
+        var stringOfPhoneNumber = String(describing: phoneNumber)
+        if stringOfPhoneNumber.characters.count == 10 {
+            let startIndex = stringOfPhoneNumber.startIndex
+            stringOfPhoneNumber.insert("(", at: startIndex)
+            stringOfPhoneNumber.insert(")", at: stringOfPhoneNumber.index(startIndex, offsetBy: 4))
+            stringOfPhoneNumber.insert(" ", at: stringOfPhoneNumber.index(startIndex, offsetBy: 5))
+            stringOfPhoneNumber.insert("-", at: stringOfPhoneNumber.index(startIndex, offsetBy: 9))
+            
+            return stringOfPhoneNumber
+        } else{
+            
+            return stringOfPhoneNumber
+        }
+    }
+    
+    //MARK: tableview functions
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+
+        return 2
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if section == 1 {
+            return emergencyArray.count
+        }
+        return 1
+    }
+    
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+
+        //create everything for first cell
+        if indexPath.section == 0 {
+            let cell = UITableViewCell()
```

Comment:
Dequeue your cells! Don't create your own cells or else your performance will go to shit along with your memory footprint! The cell creation of adding subviews and all of this shouldn't be handled upon creation as that's also super slow if trying to maintain good fps.